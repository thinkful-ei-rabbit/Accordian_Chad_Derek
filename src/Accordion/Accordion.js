import React, { Component } from 'react';

class Accordion extends Component {

  static defaultProps = {
    sections: []
  }

  state = {
    index: null,
    display:''
  }

  handleClick(index) {
    if (this.state.index !== index){
      this.setState({
        index: index,
        display: this.props.sections[index].content
      })
    } else {
      this.setState({
        index: null,
        display: ''
      })
    }
  }

  render() {
    const buttons = this.props.sections.map((section, index) => (
      <li>
        <button
        onClick={() => this.handleClick(index)}
        key={index}
        >
          {section.title}
        </button>
      </li>
    ))
    
    return <div>
      <ul>
        {buttons}
      </ul>
      <p>{this.state.display}</p>
    </div>
  }
}


export default Accordion;