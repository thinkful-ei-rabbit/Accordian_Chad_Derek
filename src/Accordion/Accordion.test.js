import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const sectionsProp = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]


describe('Accordion test suite', () =>{

  it.skip('Snap 1', () => {
    const tree = renderer
    .create(<Accordion />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it.skip('Snap 2', () => {
    const tree = renderer
    .create(<Accordion sections={[
      {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
      }
    ]}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it.skip('opens any clicked section', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Snap 4', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
})