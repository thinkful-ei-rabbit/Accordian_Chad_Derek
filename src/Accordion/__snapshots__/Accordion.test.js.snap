// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion test suite Snap 1 1`] = `
<div>
  <ul />
  <p>
    
  </p>
</div>
`;

exports[`Accordion test suite Snap 2 1`] = `
<div>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <p>
    
  </p>
</div>
`;

exports[`Accordion test suite Snap 4 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <p>
    Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
  </p>
</div>
`;

exports[`Accordion test suite opens any clicked section 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <p>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </p>
</div>
`;
